audb/core/publish.py

```diff
@@ -104,7 +104,14 @@ def _find_attachments(
     # release dependencies to removed attachment files
     db_attachment_files = []
     for attachment_id in list(db.attachments):
-        db_attachment_files += db.attachments[attachment_id].files
+        files = db.attachments[attachment_id].files
+        if len(files) == 0:
+            raise RuntimeError(
+                'You need to include at least one file '
+                'when using a folder as attachment '
+                f"as in '{attachment_id}'."
```

Comment:
```suggestion
                "An attached folder must "
                "contain at least one file. "
                f"But attachment '{attachment_id}' "
                f"points to an empty folder."
```

Reply:
The folder don't have to be empty to raise this error, it could also contain sub-folders.

I updated the message to:

```python
                "An attached folder must "
                "contain at least one file. "
                f"But attachment '{attachment_id}' "
                f"doesn't contain any files."
```

Reply:
BTW, if you create the following structure for your attachments:

```
folder-a/
    folder-b/
    folder-c/file.txt
```
publish it and load it you would get back the following:
```
folder-a/
    folder-c/file.txt
```

Reply:
Question is if we should raise a warning during publication, or if we should simply document this behavior in the docstring of `publish()`?

Reply:
Good point. Since we almost never work with warnings, I would either raise an error or document it.

Reply:
I raise now an error if the attachment contains an empty sub-folder.